src/plugins/mf_priority.cpp

```diff
@@ -264,7 +292,56 @@ static int validate_cb (flux_plugin_t *p,
                                  NULL) < 0)
         flux_log_error (h, "flux_jobtap_job_aux_set");
 
-    bank_it->second.current_jobs++;
+    return 0;
+}
+
+
+static int depend_cb (flux_plugin_t *p,
+                      const char *topic,
+                      flux_plugin_arg_t *args,
+                      void *data)
+{
+    int userid;
+    long int id;
+    struct bank_info *b;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:I}",
+                                "userid", &userid, "id", &id) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    b = static_cast<bank_info *> (flux_jobtap_job_aux_get (
+                                                    p,
+                                                    FLUX_JOBTAP_CURRENT_JOB,
+                                                    "mf_priority:bank_info"));
+
+    if (b == NULL) {
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB, "plugin",
+                                     3, "mf_priority - job.state.depend: " \
+                                     "bank info is missing");
+
+        return -1;
+    }
+
+    // if user has already hit their max running jobs count, add a job
+    // dependency to hold job until an already running job has finished
+    if ((b->max_running_jobs > 0) && (b->current_jobs == b->max_running_jobs)) {
+        if (flux_jobtap_dependency_add (p, id, "max-jobs-limit") < 0) {
+            flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB, "plugin",
```

Comment:
Same here (fatal exception) If we can't add a dependency to the job then something bad has happened and we shouldn't let the job continue.